Issue when using mini_racer as JavaScript runtime to deploy a Rails 6 app with Capistrano

Hi, 
I have an issue when deploying a Rails 6.0.3.3 app with Capistrano.

This is the error I get on assets:precompile task:

```
...
Tasks: TOP => deploy:assets:precompile
(See full trace by running task with --trace)
The deploy has failed with an error: Exception while executing as {USER}@{SERVER}: rake exit status: 1
rake stdout: rake aborted!
ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes.
...
```

It seems that ExecJS is not able to detect mini_racer runtime.

_App relevant gems_
mini_racer (0.3.1)
libv8 (8.4.255.0)

_Server_
Apache 2.4.x
Ruby 2.6.6
Passenger 6.0.6

**Locking libv8 version to 7.x and downgrading mini_racer to version 0.2.15 solves the issue.**

Many thanks in advance for your help.

Mauro

Hi folks, I tried to install mini_racer 0.4.0 but the issue persists.

The libv8-node 15.14.0.0 installs successfully, the problem occurs while installing mini_racer.

Relevant info:
OS: Red Hat Enterprise Linux Server release 7.9 (Maipo)
GCC: 4.8.5
Ruby: 2.6.6 (rvm)

Many thanks in advance for your help.
Mauro


Here is the log:

```
current directory: /usr/local/rvm/gems/ruby-2.6.6/gems/mini_racer-0.4.0/ext/mini_racer_extension
/usr/local/rvm/rubies/ruby-2.6.6/bin/ruby -I /usr/local/rvm/rubies/ruby-2.6.6/lib/ruby/site_ruby/2.6.0 -r ./siteconf20210413-9105-fanf3q.rb extconf.rb
checking for -lpthread... yes
creating Makefile

current directory: /usr/local/rvm/gems/ruby-2.6.6/gems/mini_racer-0.4.0/ext/mini_racer_extension
make "DESTDIR=" clean

current directory: /usr/local/rvm/gems/ruby-2.6.6/gems/mini_racer-0.4.0/ext/mini_racer_extension
make "DESTDIR="
compiling mini_racer_extension.cc
cc1plus: warning: command line option ‘-Wimplicit-int’ is valid for C/ObjC but not for C++ [enabled by default]
In file included from /usr/local/rvm/rubies/ruby-2.6.6/include/ruby-2.6.0/ruby/ruby.h:29:0,
                 from /usr/local/rvm/rubies/ruby-2.6.6/include/ruby-2.6.0/ruby.h:33,
                 from mini_racer_extension.cc:2:
/usr/local/rvm/rubies/ruby-2.6.6/include/ruby-2.6.0/ruby/defines.h:388:44: warning: invalid suffix on literal; C++11 requires a space between literal and identifier [-Wliteral-suffix]
 #   define EXTERN __pragma(message(__FILE__"("STRINGIZE(__LINE__)"): warning: "\
                                            ^
In file included from /usr/local/rvm/rubies/ruby-2.6.6/include/ruby-2.6.0/ruby/intern.h:24:0,
                 from /usr/local/rvm/rubies/ruby-2.6.6/include/ruby-2.6.0/ruby/ruby.h:2111,
                 from /usr/local/rvm/rubies/ruby-2.6.6/include/ruby-2.6.0/ruby.h:33,
                 from mini_racer_extension.cc:2:
/usr/local/rvm/rubies/ruby-2.6.6/include/ruby-2.6.0/ruby/defines.h:388:44: warning: invalid suffix on literal; C++11 requires a space between literal and identifier [-Wliteral-suffix]
 #   define EXTERN __pragma(message(__FILE__"("STRINGIZE(__LINE__)"): warning: "\
                                            ^
In file included from mini_racer_extension.cc:5:0:
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h: In member function ‘void v8::TracedReferenceBase<T>::SetSlotThreadSafe(T*)’:
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:912:22: error: expected type-specifier
     reinterpret_cast<std::atomic<T*>*>(&val_)->store(new_val,
                      ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:912:22: error: expected ‘>’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:912:22: error: expected ‘(’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:912:22: error: ‘atomic’ is not a member of ‘std’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:912:35: error: expected primary-expression before ‘*’ token
     reinterpret_cast<std::atomic<T*>*>(&val_)->store(new_val,
                                   ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:912:36: error: expected primary-expression before ‘>’ token
     reinterpret_cast<std::atomic<T*>*>(&val_)->store(new_val,
                                    ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:912:38: error: expected primary-expression before ‘>’ token
     reinterpret_cast<std::atomic<T*>*>(&val_)->store(new_val,
                                      ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:913:54: error: ‘memory_order_relaxed’ is not a member of ‘std’
                                                      std::memory_order_relaxed);
                                                      ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:913:80: error: expected ‘)’ before ‘;’ token
                                                      std::memory_order_relaxed);
                                                                                ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h: In member function ‘const T* v8::TracedReferenceBase<T>::GetSlotThreadSafe() const’:
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:29: error: expected type-specifier
     return reinterpret_cast<std::atomic<const T*> const*>(&val_)->load(
                             ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:29: error: expected ‘>’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:29: error: expected ‘(’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:29: error: ‘atomic’ is not a member of ‘std’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:41: error: expected primary-expression before ‘const’
     return reinterpret_cast<std::atomic<const T*> const*>(&val_)->load(
                                         ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:41: error: expected ‘)’ before ‘const’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:41: error: expected ‘;’ before ‘const’
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:49: error: expected unqualified-id before ‘>’ token
     return reinterpret_cast<std::atomic<const T*> const*>(&val_)->load(
                                                 ^
/usr/local/rvm/gems/ruby-2.6.6/gems/libv8-node-15.14.0.0-x86_64-linux/vendor/v8/include/v8.h:920:49: error: expected initializer before ‘>’ token
At global scope:
cc1plus: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
cc1plus: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1plus: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1plus: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1plus: warning: unrecognized command line option "-Wno-cast-function-type" [enabled by default]
make: *** [mini_racer_extension.o] Error 1

make failed, exit code 2
```

--- a/README.md
+++ b/README.md
@@ -416,14 +416,14 @@ Or install it yourself as:
     $ gem install mini_racer
 
 
-**Note** using v8.h and compiling MiniRacer requires a C++11 standard compiler, more specifically clang 3.5 (or later) or gcc 4.8 (or later).
+**Note** using v8.h and compiling MiniRacer requires a C++11 standard compiler, more specifically clang 3.5 (or later) or GCC 6.3 (or later).
 
 
 ## Travis-ci
 
-To install `mini-racer` you will need a version of gcc that supports C++11 (gcc 4.8) this is included by default in ubuntu trusty based images.
+To install `mini-racer` you will need a version of GCC that supports C++11 (GCC 6.3) this is included by default in ubuntu trusty based images.
 
-Travis today ships by default with a precise based image. Precise Pangolin (12.04 LTS) was first released in August 2012. Even though you can install GCC 4.8 on precise the simpler approach is to opt for the trusty based image.
+Travis today ships by default with a precise based image. Precise Pangolin (12.04 LTS) was first released in August 2012. Even though you can install GCC 6.3 on precise the simpler approach is to opt for the trusty based image.
 
 Add this to your .travis.yml file:
 

